### PR TITLE
ocamlPackages.semver: init at 0.1.0

### PR DESCRIPTION
**Pull request by ulrikstrid**
###### Description of changes

Broken out from #195344

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mgttlinger** [comment]
Thanks for submitting the package! Two things that I notice: 
1. There is [specific build support for OCaml packages](https://nixos.org/manual/nixpkgs/stable/#sec-language-ocaml-packaging) that would clean up this derivation a bit. The manual only lists `buildDunePackage` however there is also a `buildOasisPackage` with the same call semantic
2. There is `fetchFromGitHub` (which you can also see in use in the linked manual page) that would be a bit cleaner than fetching the tarball

**Comment by ulrikstrid** [comment]
Thank you @mgttlinger for the review, I made the changes and it still seems to work and is cleaner :+1:

**Review by mgttlinger — APPROVED** [review]
Changes look good and the package builds. Thanks!

File: pkgs/development/ocaml-modules/semver/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ lib, fetchFromGitHub, buildOasisPackage }:
+
+buildOasisPackage rec {
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/semver/default.nix:3]: Sounds overkill. There is no need for `oasis` here.

Comment (ulrikstrid) [reply pkgs/development/ocaml-modules/semver/default.nix:3]: So I'll revert to mkDerivation?

Comment (vbgl) [reply pkgs/development/ocaml-modules/semver/default.nix:3]: Commit f84ac1e912a6f408060162c9ef88a3ffbe031839 is nearly good: it misses a `meta.platforms` and `ocaml${ocaml.version}` as part of the `pname`.

Comment (ulrikstrid) [reply pkgs/development/ocaml-modules/semver/default.nix]: Should be OK now then

File: pkgs/development/ocaml-modules/semver/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "ocaml${ocaml.version}-semver";
+  version = "0.1.0";
+  src = fetchzip {
+    url = "https://github.com/rgrinberg/ocaml-semver/archive/v${version}.tar.gz";
+    sha256 = "sha256-0BzeuVTpuRIQjadGg08hTvMzZtKCl2utW2YK269oETk=";
+  };
+
+  minimalOCamlVersion = "4.02";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/semver/default.nix:11]: This attribute is ignored by `mkDerivation`. Can you please use an assertion (e.g., with `lib.throwIfNot`) instead?

**Pull request merged**